### PR TITLE
fix: Empty / NotEmpty operators for descriptions, tags [WEB-1751]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

The experiment filters for empty / not-empty were queries for NULL / IS NOT NULL
For description we should also consider an empty string
For tags we should consider the string "[]" to be empty

## Test Plan

Filter a project's experiments by description empty / is not empty
Filter a project's experiments by tags empty / is not empty

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/experiment_filter.go [hunk 1]
```diff
@@ -361,9 +361,16 @@ func (e experimentFilter) toSQL(q *bun.SelectQuery,
 			case doesNotContain:
 				queryString = "? NOT ILIKE ?"
 				queryArgs = append(queryArgs, bun.Safe(col), fmt.Sprintf("%%%s%%", *e.Value))
-			case empty, notEmpty:
-				queryString = "? ?"
+			case empty:
+				queryString = "? ? OR ? ? ? OR ? ? ?"
+				queryArgs = append(queryArgs, bun.Safe(col), bun.Safe(oSQL))
```
Comment (stoksc) [inline master/internal/experiment_filter.go:366]: once you are in this empty case, just writing out the fragment is significantly more readable to me
```suggestion
				queryString = "? IS NULL OR ? = "" OR ? = []"
				queryArgs = append(queryArgs, bun.Safe(col), bun.Safe(col), bun.Safe(col))
```

Comment (stoksc) [reply master/internal/experiment_filter.go:366]: i know that goes against what is happening here already, so i'm not blocking, just giving my 2c.

**Review by stoksc — APPROVED** [review]
looks like it should work to me

File: master/internal/experiment_filter.go [hunk 2]
```diff
@@ -361,9 +361,16 @@ func (e experimentFilter) toSQL(q *bun.SelectQuery,
...
+				queryArgs = append(queryArgs, bun.Safe(col), bun.Safe(equal), "")
```
Comment (stoksc) [inline master/internal/experiment_filter.go:367]: total nit, it looks like `equal` and `notEqual` are used to match on the input format, so using them in the output query is maybe unintended coupling
```suggestion
				queryArgs = append(queryArgs, bun.Safe(col), bun.Safe(equal), "")
```

File: master/internal/experiment_filter.go (hunk 1 above)
Comment (mapmeld) [reply master/internal/experiment_filter.go:366]: this does look a lot neater ✅

**Pull request merged**